src/main/java/hlf/java/rest/client/service/impl/NetworkStatusImpl.java

```diff
@@ -90,67 +98,92 @@ public ResponseEntity<ClientResponseModel> getChannelFromNetwork(String channelN
         HttpStatus.OK);
   }
 
+  private ConfigUpdate getChannelConfig(
+      String channelName, ChannelUpdateParamsDTO organizationDetails) {
+    ConfigUpdate.Builder configUpdateBuilder = createConfigUpdate(channelName);
+    return configUpdateBuilder
+        .setWriteSet(
+            updateChannel.buildWriteset(configUpdateBuilder.getReadSet(), organizationDetails))
+        .build();
+  }
+
   @Override
   public ResponseEntity<ClientResponseModel> generateConfigUpdate(
       String channelName, NewOrgParamsDTO organizationDetails) {
     Network network = gateway.getNetwork(channelName);
 
+    if (network != null) {
+      String base64EncodedByteArrayDeserialized =
+          getDeserializedConfig(getChannelConfig(channelName, organizationDetails));
+      return new ResponseEntity<>(
+          new ClientResponseModel(ErrorConstants.NO_ERROR, base64EncodedByteArrayDeserialized),
+          HttpStatus.OK);
+    } else {
+      log.warn(
+          "Error generating the Config Update: Network and User cannot be NULL: "
+              + "Network = "
+              + network);
+      return new ResponseEntity<>(
+          new ClientResponseModel(
+              ErrorCode.NOT_FOUND.getValue(),
+              "Network and User cannot be NULL: " + "Network = " + network),
+          HttpStatus.OK);
+    }
+  }
+
+  private ConfigUpdate.Builder createConfigUpdate(String channelName) {
+    Network network = gateway.getNetwork(channelName);
     if (network != null) {
       try {
         Channel selectedChannel = network.getChannel();
-        ConfigUpdate selectedChannelConfigUpdate =
-            ConfigUpdate.parseFrom(selectedChannel.getChannelConfigurationBytes());
-        // ReadSet must have version, all other fields ignored; WriteSet must have changes
-        ConfigGroup readset = selectedChannelConfigUpdate.getReadSet();
-        // ConfigGroups consist of: groups, modPolicy, policies, values, and version.
-        ConfigUpdate configUpdate =
-            ConfigUpdate.newBuilder()
-                .setChannelId(channelName)
-                .setReadSet(readset)
-                .setWriteSet(
-                    addOrgToChannelWriteSetBuilder.buildWriteset(readset, organizationDetails))
-                .build();
-        MessageOrBuilder message = configUpdate;
-        String channelConfigString = JsonFormat.printer().print(message);
-        log.info(channelConfigString);
-        String base64EncodedByteArrayDeserialized =
-            Base64.getEncoder().encodeToString(channelConfigString.getBytes());
-        return new ResponseEntity<>(
-            new ClientResponseModel(ErrorConstants.NO_ERROR, base64EncodedByteArrayDeserialized),
-            HttpStatus.OK);
+        byte[] channelConfigBytes = selectedChannel.getChannelConfigurationBytes();
+        if (channelConfigBytes != null) {
+          ConfigUpdate selectedChannelConfigUpdate = ConfigUpdate.parseFrom(channelConfigBytes);
```

Comment:
Will this give us the updated config after any channel update operation or 
```
setWriteSet(
                    addOrgToChannelWriteSetBuilder.buildWriteset(readset, organizationDetails))
```
is this step required here also? As this method will help us fetch the updated config which needs to be shared for signing purpose

Reply:
Adding the writeSet to the config update is handled as part of this `getChannelConfig` method, which is called at all the places - even for generateConfigUpdate. So getChannelConfig is what we will use. 
Now I am thinking about it, I have removed the `getChannelConfig` and added the writeSet logic to `createConfigUpdate` itself and using it everywhere. Combined the logic to this one method.